.github/workflows/ci.yaml

```diff
@@ -44,7 +44,12 @@ jobs:
           sudo pip3 install yamllint
 
       - name: Run linters
-        run: ./hack/pre-commit.sh
+        run: |
+           ./hack/pre-commit.sh
+           echo "Checking that generated files were not modified"
+           make generate
+           git --no-pager diff
+           git diff-index --quiet HEAD
```

Comment:
This should be a separate step:

```
- name: "Checking that generated files were not modified"
  run: |
    make generate
    git --no-pager diff
    git diff-index --quiet HEAD
```

This makes it easy to move the step to another job later, keep the code cleaner, and make it easier to understand failures.